staging/src/k8s.io/apiserver/pkg/storage/feature/feature_support_checker.go

```diff
@@ -82,26 +86,56 @@ func (f *defaultFeatureSupportChecker) Supports(feature storage.Feature) (bool,
 func (f *defaultFeatureSupportChecker) CheckClient(ctx context.Context, c client, feature storage.Feature) error {
 	switch feature {
 	case storage.RequestWatchProgress:
-		return f.clientSupportsRequestWatchProgress(ctx, c)
+		return f.checkClient(ctx, c)
 	default:
 		return fmt.Errorf("feature %q is not implemented in DefaultFeatureSupportChecker", feature)
 
 	}
 }
 
-func (f *defaultFeatureSupportChecker) clientSupportsRequestWatchProgress(ctx context.Context, c client) error {
+func (f *defaultFeatureSupportChecker) checkClient(ctx context.Context, c client) error {
+	// start with 10 ms, multiply by 2 each step, 16 steps ~= 10 minutes
+	backoff := wait.Backoff{
+		Duration: 10 * time.Millisecond,
+		Jitter:   0,
+		Factor:   2,
+		Steps:    16,
```

Comment:
IIRC, Marek was suggesting that we shouldn't limit to 16 steps here...

@serathius - for thoughts

Reply:
My expectation is that we will continue retries every couple of seconds (like 15s) infinitely.

So the backoff can start from some low period (like 10 milisecond), grow exponentially until period of 15s, and they stay on that period.

Reply:
If i remember correctly there is a `cap` for that purpose, let me test it out and update to that behavior.